lib/apiservers/engine/backends/container.go

```diff
@@ -367,48 +438,124 @@ func (c *Container) ContainerStop(name string, seconds int) error {
+	}
+
+	// HACK: Just set a bunch of dummy info needed to get docker attach working
+	base := &types.ContainerJSONBase{
+		State: &types.ContainerState{Status: "running",
+			Running: true,
+			Paused:  false,
+		},
+	}
+
+	conJSON := &types.ContainerJSON{ContainerJSONBase: base, Config: vc.Config}
+
+	log.Printf("ContainerInspect json config = %+v\n", conJSON.Config)
+
+	return conJSON, nil
 }
 
+// ContainerLogs hooks up a container's stdout and stderr streams
+// configured with the given struct.
 func (c *Container) ContainerLogs(name string, config *backend.ContainerLogsConfig, started chan struct{}) error {
 	return fmt.Errorf("%s does not implement container.ContainerLogs", c.ProductName)
 }
 
+// ContainerStats writes information about the container to the stream
+// given in the config object.
 func (c *Container) ContainerStats(name string, config *backend.ContainerStatsConfig) error {
 	return fmt.Errorf("%s does not implement container.ContainerStats", c.ProductName)
 }
 
+// ContainerTop lists the processes running inside of the given
+// container by calling ps with the given args, or with the flags
+// "-ef" if no args are given.  An error is returned if the container
+// is not found, or is not running, or if there are any problems
+// running ps, or parsing the output.
 func (c *Container) ContainerTop(name string, psArgs string) (*types.ContainerProcessList, error) {
 	return nil, fmt.Errorf("%s does not implement container.ContainerTop", c.ProductName)
 }
 
+// Containers returns the list of containers to show given the user's filtering.
 func (c *Container) Containers(config *types.ContainerListOptions) ([]*types.Container, error) {
 	return nil, fmt.Errorf("%s does not implement container.Containers", c.ProductName)
 }
 
 // docker's container.attachBackend
 
-func (c *Container) ContainerAttach(name string, cac *backend.ContainerAttachConfig) error {
-	return fmt.Errorf("%s does not implement container.ContainerAttach", c.ProductName)
+// ContainerAttach attaches to logs according to the config passed in. See ContainerAttachConfig.
+func (c *Container) ContainerAttach(prefixOrName string, ca *backend.ContainerAttachConfig) error {
+	defer trace.End(trace.Begin("ContainerAttach"))
+
+	vc := viccontainer.GetCache().GetContainerByName(prefixOrName)
+
+	if vc == nil {
+		//FIXME: If we didn't find in the cache, we should goto the port layer and
+		//see if it exists there.  API server might have been bounced.  For now,
+		//just return error
+
+		return derr.NewRequestNotFoundError(fmt.Errorf("No such container: %s", prefixOrName))
+	}
+
+	clStdin, clStdout, clStderr, err := ca.GetStreams()
+
+	if !vc.Config.Tty && ca.MuxStreams {
+		// replace the stdout/stderr with Docker's multiplex stream
+		clStderr = stdcopy.NewStdWriter(clStderr, stdcopy.Stderr)
+		clStdout = stdcopy.NewStdWriter(clStdout, stdcopy.Stdout)
+	}
+
+	if !ca.UseStdin {
+		clStdin = nil
+	}
+	if !ca.UseStdout {
+		clStdout = nil
```

Comment:
It's intentional.  We can check these options in 543.  I debated on which would be cleaner to look at.  Maybe I'll move it into the code above.